src/Npgsql/NpgsqlConnector.cs

```diff
@@ -486,6 +489,18 @@ internal async Task Open(NpgsqlTimeout timeout, bool async, CancellationToken ca
                             Log.Error("Exception bubbled out of multiplexing read loop", t.Exception!, Id);
                         }, TaskContinuationOptions.OnlyOnFaulted);
                 }
+
+                if (_isKeepAliveEnabled)
+                {
+                    // Start the keep alive mechanism to work by scheduling the timer.
+                    // Otherwise, it doesn't work for cases when no query executed during
+                    // the connection lifetime in case of a new connector.
```

Comment:
nit: I don't think this comment is necessary, everything's pretty self-explanatory :)

Reply:
It's a reminder why it was added now. Probably keep alive should be cleaned a bit.